builtin/logical/database/path_rotate_credentials.go

```diff
@@ -75,8 +75,16 @@ func pathRotateRootCredentials(b *databaseBackend) []*framework.Path {
 }
 
 func (b *databaseBackend) pathRotateRootCredentialsUpdate() framework.OperationFunc {
-	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (resp *logical.Response, err error) {
 		name := data.Get("name").(string)
+		defer func() {
+			if err == nil {
+				b.dbEvent(ctx, "rotate-root", req.Path, name, true)
+			} else {
+				b.dbEvent(ctx, "rotate-root-fail", req.Path, name, false)
```

Comment:
The failure modes are unlikely, but I think it should be possible for modified to be true in this case, if for example the write to storage fails after the password successfully got updated. I think the same is true for a couple of other APIs as well.